src/Core/Workspace/Workspace.cs

```diff
@@ -146,17 +148,25 @@ public string CacheFolder
         /// </summary>
         public string CacheFolder { get; set; }
 
+        /// <summary>
+        /// An event that is set after the workspace initialization has completed.
+        /// </summary>
+        private ManualResetEvent initialized = new ManualResetEvent(false);
+
+        /// <inheritdoc/>
+        public Task Initialization => Task.Run(() => initialized.WaitOne());
+
```

Comment:
Is this created before calling the constructor or after? Just making sure.

Reply:
The `initialized` event above is a `ManualResetEvent` object which is created _before_ the constructor is called, and the event does not get set until the asynchronous initialization has completed. It never gets reset; once it's set, it's set forever.

As for `Initialization`, since it uses this expression-bodied property accessor via `=>`, it's actually creating a new `Task` for each caller who accesses `Workspace.Initialization`. The task is very lightweight; it just waits for the `initialized` event to be set (or, if it is already set, it just completes immediately).